server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -315,16 +322,27 @@ static ClusterState innerDelete(DeleteSearchPipelineRequest request, ClusterStat
         return newState.build();
     }
 
-    public SearchRequest transformRequest(SearchRequest searchRequest) {
-        String pipelineId = searchRequest.pipeline();
+    public SearchRequest transformRequest(SearchRequest originalRequest) {
+        String pipelineId = originalRequest.pipeline();
         if (pipelineId != null) {
             PipelineHolder pipeline = pipelines.get(pipelineId);
             if (pipeline == null) {
                 throw new IllegalArgumentException("Pipeline " + pipelineId + " is not defined");
             }
-            return pipeline.pipeline.transformRequest(searchRequest);
+            if (pipeline.pipeline.getSearchRequestProcessors().size() > 0) {
```

Comment:
the possibility here can be pipeline.pipeline.getSearchRequestProcessors() == null. we should we CollectionUtils.isEmpty to check if the list is null or empty rather than checking the size. It will make sure that if processor list is null we don't get a NPE.

Reply:
The way we initialize it, it will never be null. 

Even if no `request_processors` property is specified, we initialize to an empty list.